smac_planner/src/a_star.cpp

```diff
@@ -349,6 +366,104 @@ bool AStarAlgorithm<NodeT>::isGoal(NodePtr & node)
   return node == getGoal();
 }
 
+template<>
+AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(
+  const NodePtr & node,
+  const NodeGetter & node_getter)
+{
+  std::vector<std::pair<NodePtr, Coordinates>> possible_nodes;
+  const NodePtr & goal = getGoal();
+  ompl::base::ScopedState<> from(node->motion_table.state_space), to(
+    node->motion_table.state_space), s(node->motion_table.state_space);
+  const NodeSE2::Coordinates & node_coords = node->pose;
+  const NodeSE2::Coordinates & goal_coords = goal->pose;
```

Comment:
You can just use `_goal_coordinates` and remove the `getGoal` method call and `goal` objects.

Reply:
Fixed